robot_smach_states/src/robot_smach_states/navigation/follow_operator2_0.py

```diff
@@ -246,98 +224,79 @@ def execute(self, userdata):
+                end = int(length / self._resolution)
 
                 for i in range(start, end):
-                    x = previous_point.x() + i * dx_norm * res
-                    y = previous_point.y() + i * dy_norm * res
+                    x = previous_point.x() + i * dx_norm * self._resolution
+                    y = previous_point.y() + i * dy_norm * self._resolution
                     kdl_pose = kdl_conversions.kdl_frame_stamped_from_XYZRPY(x=x, y=y, z=0, yaw=yaw)
                     crumb_waypoint.waypoint = kdl_conversions.kdl_frame_stamped_to_pose_stamped_msg(kdl_pose)
-
             previous_point = copy.deepcopy(crumb._pose.p)
 
-        # Delete the elements from the plan within the operator radius from the robot
-        #cutoff = int(self._operator_radius / (2.0 * res))
-        #if len(kdl_plan) > cutoff:
-        #    del kdl_plan[-cutoff:]
-
-        # ros_plan = [kdl_conversions.kdl_frame_stamped_to_pose_stamped_msg(frame) for frame in kdl_plan]
-
-        # Check if plan is valid. If not, remove invalid points from the path
         if len(self._breadcrumb) > 0:
             ros_plan = [crwp.waypoint for crwp in self._breadcrumb]
             if not self._robot.base.global_planner.checkPlan(ros_plan):
-                print "Breadcrumb plan is blocked, removing blocked points"
-                # Go through plan from operator to robot and pick the first unoccupied point as goal point
-                # ros_plan = [point for point in ros_plan if self._robot.base.global_planner.checkPlan([point])]
-                self._breadcrumb = [crwp for crwp in self._breadcrumb
-                                    if self._robot.base.global_planner.checkPlan([crwp.waypoint])]
-
-        # buffer_msg = Marker()
-        # buffer_msg.type = Marker.POINTS
-        # buffer_msg.scale.x = 0.05
-        # buffer_msg.scale.y = 0.05
-        # buffer_msg.header.stamp = rospy.get_rostime()
-        # buffer_msg.header.frame_id = "/map"
-        # buffer_msg.color.a = 1
-        # buffer_msg.color.r = 1
-        # buffer_msg.color.g = 0
-        # buffer_msg.color.b = 0
-        # #buffer_msg.lifetime = rospy.Time(.0)
-        # buffer_msg.id = 0
-        # buffer_msg.action = Marker.ADD
+                rospy.loginfo("Breadcrumb plan is blocked, removing blocked points")
+                rospy.loginfo("Breadcrumbs before removing blocked points: {}".format(len(ros_plan)))
+                ros_plan = [crwp.waypoint for crwp in self._breadcrumb
+                            if self._robot.base.global_planner.checkPlan([crwp.waypoint])]
+                rospy.loginfo("Breadcrumbs length after removing blocked points: {}".format(len(ros_plan)))
+
+            # if the new ros plan is empty, we do have an operator but there are no breadcrumbs that we can reach
+            # if not ros_plan:
+            #     return 'no_follow_bread_recovery'
+        else:
+            return 'no_follow_bread_recovery'
+
+        buffer_msg = Marker()
```

Comment:
Use helper method for this? Ideally, only use messages in the API layer, i.e., a Python file in robot skills. This improves portability